### PR TITLE
fix(sink): add cassandra batch size and fix bigquery array null

### PR DESCRIPTION
**Pull request by xxhZs**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?
in this pr, we add bigquery's `max_batch_rows`  and cassandra's `max_batch_rows` `request_timeout` to fix problems of excessive data size.
And we fix bigquery array null can't insert error,(the array null will be converted to []. )
https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [x] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [x] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

`In cassandra sink.` Add two optional option. 
`cassandra.max_batch_rows: `Number of batch rows sent at a time defaults is 512, 
It should be less than 65535 according to cassandra, and ensures that the size of each batch(max_batch_rows * size of each row) is less than 50kb (Configuration items for cassandra: batch_size_fail_threshold_in_kb)
`cassandra.request_timeout_ms: `Waiting time for each batch, default is 2000ms
it not recommended to change, if you encounter timeout time is not enough, prioritize to reduce batch size

`In bigquery sink,` Add optional option 
`bigquery.max_batch_rows:` Number of batch rows sent at a time defaults is 1024, 

Also due to this document, we will convert an array with a null value to an empty array
https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types

</details>

### CODE REVIEW

File: java/connector-node/risingwave-sink-cassandra/src/main/java/com/risingwave/connector/CassandraConfig.java [hunk 1]
```diff
@@ -93,4 +99,29 @@ public CassandraConfig withPassword(String password) {
...
     }
+
+    public Integer getMaxBatchRows() {
+        return maxBatchRows;
+    }
+
+    public CassandraConfig withMaxBatchRows(Integer maxBatchRows) {
+        if (maxBatchRows > 65536 || maxBatchRows < 1) {
+            throw new IllegalArgumentException(
+                    "cassandra.max_batch_rows must be <= 65535 and >= 1");
```
Comment (hzxa21) [inline java/connector-node/risingwave-sink-cassandra/src/main/java/com/risingwave/connector/CassandraConfig.java:110]: nits: better to include the `maxBatchRows` in the error message.

File: java/connector-node/risingwave-sink-cassandra/src/main/java/com/risingwave/connector/CassandraConfig.java [hunk 2]
```diff
@@ -93,4 +99,29 @@ public CassandraConfig withPassword(String password) {
...
+        return this;
+    }
+
+    public Integer getRequestTimeoutMs() {
+        return requestTimeoutMs;
+    }
+
+    public CassandraConfig withRequestTimeoutMs(Integer requestTimeoutMs) {
+        if (requestTimeoutMs < 1) {
+            throw new IllegalArgumentException("cassandra.request_timeout_ms must be >= 1");
```
Comment (hzxa21) [inline java/connector-node/risingwave-sink-cassandra/src/main/java/com/risingwave/connector/CassandraConfig.java:122]: nits: better to include `requestTimeoutMs` in the error message.

File: src/connector/src/sink/big_query.rs [hunk 3]
```diff
@@ -61,6 +58,13 @@ pub struct BigQueryCommon {
     pub dataset: String,
     #[serde(rename = "bigquery.table")]
     pub table: String,
+    #[serde(rename = "bigquery.max_batch_rows", default = "default_max_batch_rows")]
+    #[serde_as(as = "DisplayFromStr")]
```
Comment (hzxa21) [inline src/connector/src/sink/big_query.rs:62]: do we really need to define `serde_as` here?

File: src/connector/src/sink/encoder/json.rs [hunk 4]
```diff
@@ -192,7 +203,15 @@ fn datum_to_json_object(
     custom_json_type: &CustomJsonType,
 ) -> ArrayResult<Value> {
     let scalar_ref = match datum {
-        None => return Ok(Value::Null),
+        None => {
+            if let CustomJsonType::BigQuery = custom_json_type
+                && matches!(field.data_type(), DataType::List(_))
+            {
+                return Ok(Value::Array(vec![]));
```
Comment (hzxa21) [inline src/connector/src/sink/encoder/json.rs:210]: Can you include the BigQuery doc link as a comment here to explain why we need to do this?

**Review by hzxa21 — APPROVED** [review]
Rest LGTM!

Please don't forget to update the "Documentation" section in the PR description for PR including user facing changes. Otherwise, the doc can become outdated.

File: src/connector/src/sink/big_query.rs (hunk 3 above)
Comment (xxhZs) [reply src/connector/src/sink/big_query.rs:62]: Without this, the conversion from string to usize cannot be accomplished

Comment (hzxa21) [reply src/connector/src/sink/big_query.rs:62]: I see. I missed that part.

File: src/connector/with_options_sink.yaml [hunk 5]
```diff
@@ -17,6 +17,10 @@ BigQueryConfig:
   - name: bigquery.table
     field_type: String
     required: true
+  - name: bigquery.max_batch_rows
```
Comment (hzxa21) [inline src/connector/with_options_sink.yaml:20]: I just realized that with options for cassandra sink are not included in this yaml. Let's revisit other sinks as well to ensure all sinks are covered. We can merge this PR first.

**Pull request merged**